Implement AdventureSettings & PlayerSkin packet & Sandstone

Suremeo opened the pull request:
Changes:

- Handle the AdventureSettings packet
- Flying related functions
- Added player scaling
- Added updating player skin
- Handle the PlayerSkin packet (So other players see when you change your skin ingame)
- Implemented Sandstone

Eren5960 commented:
player.go#L359 must be start with SetScale ...

Eren5960 commented:
no problem, run `go fmt ./...` in base dragonfly path and fix formatting

Suremeo commented:
> no problem, run `go fmt ./...` in base dragonfly path and fix the formatting errors

Done

DaPigGuy commented on dragonfly/entity/state/state.go:
```diff
@@ -50,6 +50,11 @@ type UsingItem struct{}
 // OnFire makes an entity show itself as on fire.
 type OnFire struct{}
 
+// Scale sets the scale of the players geometry
```
Missing punctuation

Suremeo replied:
I don't see any issue with the punctuation on this one

T14Raptor replied:
Forgot a period

Suremeo replied:
Ah, frogot documentation is supposed to have them, fixed on all files now.

DaPigGuy commented on dragonfly/player/player.go:
```diff
@@ -354,6 +356,17 @@ func (p *Player) Speed() float64 {
 	return p.speed.Load()
 }
 
+// SetScale sets the scale of the players geometry
```
Missing punctuation

DaPigGuy commented on dragonfly/player/player.go:
```diff
@@ -354,6 +356,17 @@ func (p *Player) Speed() float64 {
+	p.updateState()
+}
+
+// Scale returns the current scale of the player
```
Missing punctuation.

DaPigGuy commented on dragonfly/player/player.go:
```diff
@@ -1718,6 +1731,9 @@ func (p *Player) State() (s []state.State) {
 	if p.OnFireDuration() > 0 {
 		s = append(s, state.OnFire{})
 	}
+	if p.scale.Load() >= 1 {
```
Scale less than 1 won't be sent

Suremeo commented:
Added lots of other stuff to this pull request

Changes:

- Handle the AdventureSettings packet
- Flying related functions
- Added player scaling
- Added updating player skin
- Handle the PlayerSkin packet (So other players see when you change your skin ingame)

Sandertv commented:
Please make sure to keep separate changes in a different PR the next time. This makes it very difficult to review and get merged.

Suremeo commented:
> Please make sure to keep separate changes in a different PR the next time. This makes it very difficult to review and get merged.

Alright, will do

T14Raptor commented:
Closing in favor of d3a8a5610c33055a9efa5b9293d0ef840e2ec5af, f255865bf8b9515a722e17653005f4cedc9ffcef, 4e0e67afd2aa2b79b3a3c20db3f37856f165286d, etc.
